### PR TITLE
feat(checkbox): support keyboard navigation

### LINKED ISSUE #1494: checkbox组件支持 tab 聚焦，space和enter选中
# Feature request 🚀

### Expected:

- No breaking changes

### PR DESCRIPTION
**Pull request by tiny-dust**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

`checkbox` support keyboard navigation.

https://github.com/varletjs/varlet/assets/49502875/7d87d867-e820-40e9-90fc-5c98fd13367e

## Issues

closed #1494.

### CODE REVIEW

File: packages/varlet-ui/src/checkbox/checkbox.less [hunk 1]
```diff
@@ -35,6 +35,7 @@
   &__wrap {
     display: inline-flex;
     flex-direction: column;
+    outline: none;
```
Comment (chouchouji) [inline packages/varlet-ui/src/checkbox/checkbox.less:38]: 这个是不是单独加个class好一些，
image
image

**Comment by chouchouji** [comment]
好像有点逻辑冲突 [varlet playground link](https://varlet.gitee.io/varlet-ui-playground/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoZmFsc2UpXG5cbmZ1bmN0aW9uIGhhbmRsZUNsaWNrKCkge1xuICBjb25zb2xlLmxvZygnY2xpY2sgY2hlY2tib3gnKVxufVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPHZhci1jaGVja2JveCB2LW1vZGVsPVwidmFsdWVcIiByZWFkb25seSBAY2xpY2s9XCJoYW5kbGVDbGlja1wiPuW9k+WJjeeahOWAvDoge3sgdmFsdWUgfX08L3Zhci1jaGVja2JveD5cbjwvdGVtcGxhdGU+XG4iLCJBcHBXcmFwcGVyLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IGluc3RhbGxWYXJsZXRVSSB9IGZyb20gJy4vdmFybGV0LXJlcGwtcGx1Z2luLmpzJ1xuXG5pbnN0YWxsVmFybGV0VUkoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsInZhcmxldC1yZXBsLXBsdWdpbi5qcyI6ImltcG9ydCBWYXJsZXRVSSBmcm9tICdAdmFybGV0L3VpJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YXJsZXRDc3MgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3N0eWxlLmNzcydcbmNvbnN0IHZhcmxldFRvdWNoRW11bGF0b3IgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3RvdWNoLWVtdWxhdG9yL2lpZmUuanMnXG5cbmF3YWl0IGFwcGVuZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIGluc3RhbGxWYXJsZXRVSSgpIHtcbiAgY29uc3QgeyBwYXJlbnQgfSA9IHdpbmRvd1xuXG4gIGNvbnN0IHN0eWxlID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3R5bGUnKVxuICBzdHlsZS5pbm5lckhUTUwgPSBgXG4gICAgYm9keSB7XG4gICAgICBtaW4taGVpZ2h0OiAxMDB2aDtcbiAgICAgIHBhZGRpbmc6IDE2cHg7XG4gICAgICBtYXJnaW46IDA7XG4gICAgICBjb2xvcjogdmFyKC0tY29sb3ItdGV4dCk7XG4gICAgICBiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1jb2xvci1ib2R5KTtcbiAgICB9XG5cbiAgICAqOjotd2Via2l0LXNjcm9sbGJhciB7XG4gICAgICBkaXNwbGF5OiBub25lO1xuICAgIH1cbiAgYFxuICBkb2N1bWVudC5oZWFkLmFwcGVuZENoaWxkKHN0eWxlKVxuXG4gIGNvbnN0IHNjcmlwdCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ3NjcmlwdCcpXG4gIHNjcmlwdC5zcmMgPSB2YXJsZXRUb3VjaEVtdWxhdG9yXG4gIGRvY3VtZW50LmJvZHkuYXBwZW5kQ2hpbGQoc2NyaXB0KVxuXG4gIGNvbnN0IHRoZW1lTWFwID0ge1xuICAgIGxpZ2h0VGhlbWU6IG51bGwsXG4gICAgZGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMuZGFyayxcbiAgICBtZDNMaWdodFRoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzTGlnaHQsXG4gICAgbWQzRGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzRGFyayxcbiAgfVxuXG4gIGNvbnN0IHRoZW1lID0gcGFyZW50LmRvY3VtZW50LmRvY3VtZW50RWxlbWVudC5nZXRBdHRyaWJ1dGUoJ3RoZW1lJylcblxuICBpZiAodGhlbWUpIHtcbiAgICBWYXJsZXRVSS5TdHlsZVByb3ZpZGVyKHRoZW1lTWFwW3RoZW1lXSlcbiAgfVxuXG4gIHdpbmRvdy5hZGRFdmVudExpc3RlbmVyKCdtZXNzYWdlJywgKHsgZGF0YSB9KSA9PiB7XG4gICAgaWYgKGRhdGEuYWN0aW9uID09PSAndGhlbWUtY2hhbmdlJykge1xuICAgICAgVmFybGV0VUkuU3R5bGVQcm92aWRlcih0aGVtZU1hcFtkYXRhLnZhbHVlXSlcbiAgICB9XG4gIH0pXG5cbiAgZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignY2xpY2snLCAoKSA9PiB7XG4gICAgd2luZG93LnBhcmVudC5kb2N1bWVudC5kaXNwYXRjaEV2ZW50KG5ldyBFdmVudCgnY2xpY2snKSlcbiAgfSlcblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShWYXJsZXRVSSlcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGFwcGVuZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9IHZhcmxldENzc1xuICAgIGxpbmsub25sb2FkID0gcmVzb2x2ZVxuICAgIGxpbmsub25lcnJvciA9IHJlamVjdFxuICAgIGRvY3VtZW50LmhlYWQuYXBwZW5kQ2hpbGQobGluaylcbiAgfSlcbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJAdmFybGV0L3VpXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3ZhcmxldC5lc20uanNcIixcbiAgICBcIkB2YXJsZXQvdWkvanNvbi9hcmVhLmpzb25cIjogXCIuL3ZhcmxldC1hcmVhLmpzXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcIlByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59In0=)

readonly状态下是可以触发点击事件的，pr是不触发的

**Comment by chouchouji** [comment]
> 好像有点逻辑冲突 [varlet playground link](https://varlet.gitee.io/varlet-ui-playground/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoZmFsc2UpXG5cbmZ1bmN0aW9uIGhhbmRsZUNsaWNrKCkge1xuICBjb25zb2xlLmxvZygnY2xpY2sgY2hlY2tib3gnKVxufVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPHZhci1jaGVja2JveCB2LW1vZGVsPVwidmFsdWVcIiByZWFkb25seSBAY2xpY2s9XCJoYW5kbGVDbGlja1wiPuW9k+WJjeeahOWAvDoge3sgdmFsdWUgfX08L3Zhci1jaGVja2JveD5cbjwvdGVtcGxhdGU+XG4iLCJBcHBXcmFwcGVyLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IGluc3RhbGxWYXJsZXRVSSB9IGZyb20gJy4vdmFybGV0LXJlcGwtcGx1Z2luLmpzJ1xuXG5pbnN0YWxsVmFybGV0VUkoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsInZhcmxldC1yZXBsLXBsdWdpbi5qcyI6ImltcG9ydCBWYXJsZXRVSSBmcm9tICdAdmFybGV0L3VpJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YXJsZXRDc3MgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3N0eWxlLmNzcydcbmNvbnN0IHZhcmxldFRvdWNoRW11bGF0b3IgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3RvdWNoLWVtdWxhdG9yL2lpZmUuanMnXG5cbmF3YWl0IGFwcGVuZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIGluc3RhbGxWYXJsZXRVSSgpIHtcbiAgY29uc3QgeyBwYXJlbnQgfSA9IHdpbmRvd1xuXG4gIGNvbnN0IHN0eWxlID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3R5bGUnKVxuICBzdHlsZS5pbm5lckhUTUwgPSBgXG4gICAgYm9keSB7XG4gICAgICBtaW4taGVpZ2h0OiAxMDB2aDtcbiAgICAgIHBhZGRpbmc6IDE2cHg7XG4gICAgICBtYXJnaW46IDA7XG4gICAgICBjb2xvcjogdmFyKC0tY29sb3ItdGV4dCk7XG4gICAgICBiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1jb2xvci1ib2R5KTtcbiAgICB9XG5cbiAgICAqOjotd2Via2l0LXNjcm9sbGJhciB7XG4gICAgICBkaXNwbGF5OiBub25lO1xuICAgIH1cbiAgYFxuICBkb2N1bWVudC5oZWFkLmFwcGVuZENoaWxkKHN0eWxlKVxuXG4gIGNvbnN0IHNjcmlwdCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ3NjcmlwdCcpXG4gIHNjcmlwdC5zcmMgPSB2YXJsZXRUb3VjaEVtdWxhdG9yXG4gIGRvY3VtZW50LmJvZHkuYXBwZW5kQ2hpbGQoc2NyaXB0KVxuXG4gIGNvbnN0IHRoZW1lTWFwID0ge1xuICAgIGxpZ2h0VGhlbWU6IG51bGwsXG4gICAgZGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMuZGFyayxcbiAgICBtZDNMaWdodFRoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzTGlnaHQsXG4gICAgbWQzRGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzRGFyayxcbiAgfVxuXG4gIGNvbnN0IHRoZW1lID0gcGFyZW50LmRvY3VtZW50LmRvY3VtZW50RWxlbWVudC5nZXRBdHRyaWJ1dGUoJ3RoZW1lJylcblxuICBpZiAodGhlbWUpIHtcbiAgICBWYXJsZXRVSS5TdHlsZVByb3ZpZGVyKHRoZW1lTWFwW3RoZW1lXSlcbiAgfVxuXG4gIHdpbmRvdy5hZGRFdmVudExpc3RlbmVyKCdtZXNzYWdlJywgKHsgZGF0YSB9KSA9PiB7XG4gICAgaWYgKGRhdGEuYWN0aW9uID09PSAndGhlbWUtY2hhbmdlJykge1xuICAgICAgVmFybGV0VUkuU3R5bGVQcm92aWRlcih0aGVtZU1hcFtkYXRhLnZhbHVlXSlcbiAgICB9XG4gIH0pXG5cbiAgZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignY2xpY2snLCAoKSA9PiB7XG4gICAgd2luZG93LnBhcmVudC5kb2N1bWVudC5kaXNwYXRjaEV2ZW50KG5ldyBFdmVudCgnY2xpY2snKSlcbiAgfSlcblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShWYXJsZXRVSSlcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGFwcGVuZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9IHZhcmxldENzc1xuICAgIGxpbmsub25sb2FkID0gcmVzb2x2ZVxuICAgIGxpbmsub25lcnJvciA9IHJlamVjdFxuICAgIGRvY3VtZW50LmhlYWQuYXBwZW5kQ2hpbGQobGluaylcbiAgfSlcbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJAdmFybGV0L3VpXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3ZhcmxldC5lc20uanNcIixcbiAgICBcIkB2YXJsZXQvdWkvanNvbi9hcmVhLmpzb25cIjogXCIuL3ZhcmxldC1hcmVhLmpzXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcIlByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59In0=)
> 
> readonly状态下是可以触发点击事件的，pr是不触发的

@haoziqaq  耗子哥，这里需要你看下

**Comment by tiny-dust** [comment]
> > 好像有点逻辑冲突 [varlet playground link](https://varlet.gitee.io/varlet-ui-playground/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoZmFsc2UpXG5cbmZ1bmN0aW9uIGhhbmRsZUNsaWNrKCkge1xuICBjb25zb2xlLmxvZygnY2xpY2sgY2hlY2tib3gnKVxufVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPHZhci1jaGVja2JveCB2LW1vZGVsPVwidmFsdWVcIiByZWFkb25seSBAY2xpY2s9XCJoYW5kbGVDbGlja1wiPuW9k+WJjeeahOWAvDoge3sgdmFsdWUgfX08L3Zhci1jaGVja2JveD5cbjwvdGVtcGxhdGU+XG4iLCJBcHBXcmFwcGVyLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IGluc3RhbGxWYXJsZXRVSSB9IGZyb20gJy4vdmFybGV0LXJlcGwtcGx1Z2luLmpzJ1xuXG5pbnN0YWxsVmFybGV0VUkoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsInZhcmxldC1yZXBsLXBsdWdpbi5qcyI6ImltcG9ydCBWYXJsZXRVSSBmcm9tICdAdmFybGV0L3VpJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YXJsZXRDc3MgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3N0eWxlLmNzcydcbmNvbnN0IHZhcmxldFRvdWNoRW11bGF0b3IgPSAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3RvdWNoLWVtdWxhdG9yL2lpZmUuanMnXG5cbmF3YWl0IGFwcGVuZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIGluc3RhbGxWYXJsZXRVSSgpIHtcbiAgY29uc3QgeyBwYXJlbnQgfSA9IHdpbmRvd1xuXG4gIGNvbnN0IHN0eWxlID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3R5bGUnKVxuICBzdHlsZS5pbm5lckhUTUwgPSBgXG4gICAgYm9keSB7XG4gICAgICBtaW4taGVpZ2h0OiAxMDB2aDtcbiAgICAgIHBhZGRpbmc6IDE2cHg7XG4gICAgICBtYXJnaW46IDA7XG4gICAgICBjb2xvcjogdmFyKC0tY29sb3ItdGV4dCk7XG4gICAgICBiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1jb2xvci1ib2R5KTtcbiAgICB9XG5cbiAgICAqOjotd2Via2l0LXNjcm9sbGJhciB7XG4gICAgICBkaXNwbGF5OiBub25lO1xuICAgIH1cbiAgYFxuICBkb2N1bWVudC5oZWFkLmFwcGVuZENoaWxkKHN0eWxlKVxuXG4gIGNvbnN0IHNjcmlwdCA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ3NjcmlwdCcpXG4gIHNjcmlwdC5zcmMgPSB2YXJsZXRUb3VjaEVtdWxhdG9yXG4gIGRvY3VtZW50LmJvZHkuYXBwZW5kQ2hpbGQoc2NyaXB0KVxuXG4gIGNvbnN0IHRoZW1lTWFwID0ge1xuICAgIGxpZ2h0VGhlbWU6IG51bGwsXG4gICAgZGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMuZGFyayxcbiAgICBtZDNMaWdodFRoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzTGlnaHQsXG4gICAgbWQzRGFya1RoZW1lOiBWYXJsZXRVSS5UaGVtZXMubWQzRGFyayxcbiAgfVxuXG4gIGNvbnN0IHRoZW1lID0gcGFyZW50LmRvY3VtZW50LmRvY3VtZW50RWxlbWVudC5nZXRBdHRyaWJ1dGUoJ3RoZW1lJylcblxuICBpZiAodGhlbWUpIHtcbiAgICBWYXJsZXRVSS5TdHlsZVByb3ZpZGVyKHRoZW1lTWFwW3RoZW1lXSlcbiAgfVxuXG4gIHdpbmRvdy5hZGRFdmVudExpc3RlbmVyKCdtZXNzYWdlJywgKHsgZGF0YSB9KSA9PiB7XG4gICAgaWYgKGRhdGEuYWN0aW9uID09PSAndGhlbWUtY2hhbmdlJykge1xuICAgICAgVmFybGV0VUkuU3R5bGVQcm92aWRlcih0aGVtZU1hcFtkYXRhLnZhbHVlXSlcbiAgICB9XG4gIH0pXG5cbiAgZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignY2xpY2snLCAoKSA9PiB7XG4gICAgd2luZG93LnBhcmVudC5kb2N1bWVudC5kaXNwYXRjaEV2ZW50KG5ldyBFdmVudCgnY2xpY2snKSlcbiAgfSlcblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShWYXJsZXRVSSlcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGFwcGVuZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9IHZhcmxldENzc1xuICAgIGxpbmsub25sb2FkID0gcmVzb2x2ZVxuICAgIGxpbmsub25lcnJvciA9IHJlamVjdFxuICAgIGRvY3VtZW50LmhlYWQuYXBwZW5kQ2hpbGQobGluaylcbiAgfSlcbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJAdmFybGV0L3VpXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdmFybGV0L3VpL2VzL3ZhcmxldC5lc20uanNcIixcbiAgICBcIkB2YXJsZXQvdWkvanNvbi9hcmVhLmpzb25cIjogXCIuL3ZhcmxldC1hcmVhLmpzXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcIlByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59In0=)
> > readonly状态下是可以触发点击事件的，pr是不触发的
> 
> @haoziqaq 耗子哥，这里需要你看下

这里我看了，是我的问题，没有和原本的行为保持一致，稍后提交代码

File: packages/varlet-ui/src/checkbox/checkbox.less (hunk 1 above)
Comment (tiny-dust) [reply packages/varlet-ui/src/checkbox/checkbox.less:38]: > 这个是不是单独加个class好一些， image image

File: packages/varlet-ui/src/checkbox/Checkbox.vue [hunk 2]
```diff
@@ -182,6 +194,40 @@ export default defineComponent({
...
+
+    function handleKeyup(event: KeyboardEvent) {
+      const { disabled } = props
+      if (!isEffectFocusing.value || form?.disabled.value || disabled) {
+        return
+      }
+
+      const { key } = event
+
+      if (key === ' ') {
```
Comment (chouchouji) [inline packages/varlet-ui/src/checkbox/Checkbox.vue:225]: 这里可以简化下，如果 `event.key  !== ' '`，直接return

File: packages/varlet-ui/src/checkbox/Checkbox.vue [hunk 3]
```diff
@@ -1,5 +1,12 @@
 <template>
-  <div :class="n('wrap')" @click="handleClick">
+  <div
+    ref="root"
+    :tabindex="disabled || formDisabled ? undefined : '0'"
+    :class="classes(n('wrap'), [isEffectFocusing, n('--outline-none')])"
```
Comment (chouchouji) [inline packages/varlet-ui/src/checkbox/Checkbox.vue:5]: outline可以确认下是否像uploader组件那样直接添加，没有条件判断

Comment (tiny-dust) [reply packages/varlet-ui/src/checkbox/Checkbox.vue:5]: > outline可以确认下是否像uploader组件那样直接添加，没有条件判断

@haoziqaq 我觉得应该需要加上条件判断，不应该直接从组件的层面上限制`outline: none;`

Comment (haoziqaq) [reply packages/varlet-ui/src/checkbox/Checkbox.vue:5]: 这里不需要加条件判断。
uploader 组件需要拆出 --outline-none 是受插槽设计的影响。这里应该不存在这个问题。

Comment (tiny-dust) [reply packages/varlet-ui/src/checkbox/Checkbox.vue:5]: > 这里不需要加条件判断。 uploader 组件需要拆出 --outline-none 是受插槽设计的影响。这里应该不存在这个问题。

ok，那我重新修改一下

**Review by chouchouji — APPROVED** [review]

**Pull request merged**